### ISSUE TITLE
detect unused functions

### ISSUE THREAD

**Issue by FinzelStefan** [koalaman/shellcheck#2126]
For new checks and feature suggestions
- [ x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Unused variables are detected by shellcheck but not functions.
bash allows "exporting" a function. Using export -f creates an environment variable with the function body. 

#!/bin/bash -f
function FUNCTION_UNUSED() {
  VAR_UNUSED=TRUE
}
#eof

Here's what I got:
Line 3:
  VAR_UNUSED=TRUE
  ^-- SC2034: VAR_UNUSED appears unused. Verify use (or export if used externally).

Here's what I wanted or expected to see:
Line 2:
function FUNCTION_UNUSED()
              ^-- SC####: FUNCTION_UNUSED () appears unused. Verify use (or export if used externally).
Line 3:
  VAR_UNUSED=TRUE
  ^-- SC2034: VAR_UNUSED appears unused. Verify use (or export if used externally).

**Comment by walksanatora** [comment]
yeah i was about to suggest this also, good thing i searched to see if someone else suggested it
